simpleviewinc/keg-hub
# Add jsutils helper methods

## Context
* I written a number of methods that I think should be added to jsUtils
* This allows them to be reused in other places as needed

## Updates
* `repos/jsutils/src/array/flatUnion.js`
  * Helper array method that just calls to other helper methods => `flatArr`, then `uniqArr`
  * Useful for passing in a bunch of arrays, flattening it down to a single array, then removing duplicates
* `repos/jsutils/src/node/findProc.js`
  * This is a node only helper method for checking if a process is currently running on the machine
*   `repos/jsutils/src/object/intersectKeys.js`
  * This method is helpful for splitting the keys of a single object into two objects with only one iteration
  * This was inspired by [this code in re-theme](https://github.com/simpleviewinc/keg-hub/blob/7d28d97b511e1162fc7db219dc6ca2b6acec7007/repos/re-theme/src/styleInjector/injectHelpers.js#L65)
  * We're looping over the style object twice, once with `omitKeys`, then again with `pickKeys`
  * If we update to `intersectKeys` instead, we'd get both with only a single iteration
* I added tests for each one of the new methods
* I cleaned up the export of the node helper methods to be consistent

## Testing
* Pull this pr =>  `keg jsutils && key pr <>`
* Install dependencies => `yarn install`
* Run the tests =>  `yarn test`
  * Ensure all the tests pass
